app/main.py

```diff
@@ -1,3 +1,49 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: int | Distance) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: int) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: int | Distance) -> Distance:
+        return Distance(self.km * other)
+
+    def __round__(self) -> int:
+        return round(self.km)
+
+    def __truediv__(self, other: float | Distance) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: float | Distance) -> bool:
+        return self.km < other
+
+    def __gt__(self, other: float | Distance) -> bool:
+        return self.km > other
+
+    def __eq__(self, other: float | Distance) -> bool:
+        return self.km == other
+
+    def __le__(self, other: float | Distance) -> bool:
+        return self.km <= other
+
+    def __ge__(self, other: float | Distance) -> bool:
+        return self.km >= other
```

Comment:
Add checks for `isinstance`

Reply:
why to check in such way, why to add redundant extra rows if the code works well and passes all the tests?

Reply:
Please explain!

Reply:
These checks are needed because you can pass Distance as well as int of float instances to this method. And the logic of comparison differs. But for your case - I checked the code locally and the tests really run successfullly and I wonder why) I need to find an explanation, need to dig in more into `__ge__` internal implementation

Reply:
Anyway, you are missing `int` in annotations, add it please